### ISSUE TITLE
[Feature Request] Python function docs support like spacemacs

### ISSUE THREAD

**Issue by jramapuram** [bbatsov/prelude#1160]
When typing a function spacemacs lists the parameters of the function in the minibuffer. This is super helpful (rather than going to the function documentation to read and then returning). Any chance of getting this included? Not sure what package(s) are used for this.

**Comment by bbatsov** [comment]
I guess we can easily do this if someone checks what exactly are they using there.

**Comment by jramapuram** [comment]
Looks like this a base feature in Jedi? See screenshots from their repo: https://github.com/davidhalter/jedi

prelude does do autocomplete:

![screenshot from 2018-04-06 18-58-21](https://user-images.githubusercontent.com/8204807/38433823-95dffd7e-39cc-11e8-8658-4691ab43c465.png)

but doesn't show any info w.r.t the functions : 

![screenshot from 2018-04-06 18-58-59](https://user-images.githubusercontent.com/8204807/38433822-95c57f26-39cc-11e8-8b98-89afc6751b96.png)

**Comment by bbatsov** [comment]
PRs welcome! :-)

**Comment by palikar** [comment]
I took a closer look at the issue and I actually managed to get it to work. The solution though is not a trivial one. The problem stems from a jedi API update that is not reflected in the [emacs-jedi](https://github.com/tkf/emacs-jedi) package. There is actually a opened [PR](https://github.com/tkf/emacs-jedi/pull/294) in the repo that addresses the issue but it's not merged yet. I myself made the changes from the [commit](https://github.com/tkf/emacs-jedi/pull/294/commits) in the `jedi-core` package and now everything work fine. I use the following configuration in my personal config:
```elisp
(add-hook 'python-mode-hook 'jedi:setup)
(add-hook 'python-mode-hook 'jedi:ac-setup)
(setq jedi:setup-keys t)
(setq jedi:complete-on-dot t)
(setq jedi:tooltip-method nil)
(setq elpy-rpc-backend "jedi")
(setq jedi:get-in-function-call-delay 0)
```
